services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java

```diff
@@ -406,6 +427,17 @@ public static Response redirectAfterSuccessfulFlow(KeycloakSession session, Real
 
     }
 
+    // If a locale has been set on the login screen, associate that locale with the user
```

Comment:
No.  LocaleHelper was not being called at this point.  Now I'm calling it and LocaleHelper will set the attribute in some cases.  But at this point during login, we need to be sure that it gets set.

IMO, LocaleHelper.getLocale() should not have the side effect of setting the Locale in the UserModel.  But I didn't want to disturb that logic because I'm sure others are depending on that behavior.